### ISSUE TITLE
docs: document subnet behavior

### ISSUE THREAD

**Issue by rogpeppe** [tailscale/tailscale#2337]
I might have missed it somewhere, but the subnet routing docs don't seem to describe what the subnet routing actually does.

After reading the docs but without trying it out, I initially thought that a command like `sudo tailscale up --advertise-routes=10.0.0.0/24,10.0.1.0/24` might map a tailscale address such as `100.x.y.z`  where `100.x.y.w` is the address of the relay node to `10.x.y.z` but that I don't think that can be right because then it can't know which subnet to address to. On the other hand, if the subnet is made available directly as a `10.*` address, how does a router on another machine know that it's a tailscale address (and what about clashes) ?

So it would be helpful to naive people like me to describe what addresses should be used to address a machine on a subnet and how the mapping works.

In general, where there's an example command, it would be great if there's a full description of what that command actually does, rather than relying on reader intuition. Here's an example:

![image](https://user-images.githubusercontent.com/66491/124596561-533f1c80-de5a-11eb-97a7-d54435864711.png)

In that description, it doesn't actually say what that command is doing (what does the `-advertise-routes` flag actually imply?). What does it mean to be a "subnet router" in the context of tailscale?

**Comment by rosszurowski** [comment]
Thanks for this suggestion. I added a diagram [to the start of the article](https://tailscale.com/kb/1019/subnets/) a while back, but it only sort-of hints at the core idea of subnet routing. It would be worth having a "How subnet routing works" or "What is subnet routing" at the top that explicitly describes this.

I suppose the core aspects are…

- Exposing a route (eg. `10.x.y.z`) and approving it in the admin panel makes it available to all the other devices on your Tailscale network
- While Tailscale is running and is logged into your network, any requests to the exposed/approved routes will be sent through Tailscale
- You can access any IPs exposed by the subnet router by just using the IP address directly
- Right now, if multiple subnet routers expose the same route, the control server picks one and serves that. There's no ability to remap
- Right now, if a subnet overlaps with a local route for your device (eg. `192.168.0.0/24`), _I believe_ the Tailscale subnet overrides it
- Devices can opt-out of using subnets if needed. Instructions differ by OS

I do think a few small additions, whether in words or pictures, could go a long way to making this concept more clear.

**Comment by Oujiii** [comment]
> * Right now, if a subnet overlaps with a local route for your device (eg. `192.168.0.0/24`), _I believe_ the Tailscale subnet overrides it

Yes, just to add on this, this does happen. Whenever I have my own subnet advertised, the other nodes accepting it are unable to connect to the internet (but are still reacheable through tailscale), at least the Linux machines. I have to delete the routing manually to solve this.

I might create a new issue just for this, as a bug report.

**Comment by rogpeppe** [comment]
See also #15104
